### PR TITLE
EOS-24939: Use file lock to prevent two s3_setup process writing same…

### PR DESCRIPTION
**Pull request by t7ko-seagate**
… files in parallel

Signed-off-by: Ivan Tishchenko <ivan.tishchenko@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: scripts/provisioning/s3_setup [hunk 1]
```diff
@@ -88,7 +89,10 @@ def main():
     if args.command == 'post_install':
       from postinstallcmd import PostInstallCmd
       logger.info('post_install started.')
-      PostInstallCmd(args.config, args.services).process()
+      with open('/etc/cortx/s3/post_install.lock', 'w' as lock:
+        fcntl.flock(lock, fcntl.LOCK_EX)
+        PostInstallCmd(args.config, args.services).process()
```
Comment (bkirunge7) [inline scripts/provisioning/s3_setup:94]: id there chance PostInstallCmd does system exit, how lock is released

Comment (nileshgovande) [reply scripts/provisioning/s3_setup:94]: Add logs

File: scripts/provisioning/s3_setup [hunk 2]
```diff
@@ -104,8 +105,15 @@ def main():
         configure_only_openldap = True
       elif args.only_haproxy == True:
         configure_only_haproxy = True
-      ConfigCmd(args.config, args.services).process(configure_only_openldap,
-                                configure_only_haproxy)
+      lockfile = '/etc/cortx/s3/s3_setup.lock'
```
Comment (nileshgovande) [inline scripts/provisioning/s3_setup:108]: Pick this from confstore

Comment (nileshgovande) [reply scripts/provisioning/s3_setup:108]: ('CONFIG>CONFSTORE_BASE_CONFIG_PATH')

**Pull request merged**